crates/common/consensus/src/misc.rs

```diff
@@ -0,0 +1,158 @@
+        let pivot = bytes_to_int64(&hash_with_round(seed, round)[..]) % index_count as u64;
+        index = do_round(seed, index, pivot as usize, round, index_count);
+    }
+    Ok(index)
+}
+
+fn hash_with_round(seed: B256, round: u8) -> Hash256 {
+    let mut context = Context::new();
+
+    context.update(&seed.0);
+    context.update(&[round]);
+
+    let digest = context.finalize();
+    Hash256::from_slice(digest.as_ref())
+}
+
+fn bytes_to_int64(slice: &[u8]) -> u64 {
+    let mut bytes = [0; 8];
+    bytes.copy_from_slice(&slice[0..8]);
+    u64::from_le_bytes(bytes)
+}
+
+fn do_round(seed: B256, index: usize, pivot: usize, round: u8, list_size: usize) -> usize {
+    let flip = (pivot + (list_size - index)) % list_size;
+    let position = max(index, flip);
+    let source = hash_with_round_and_position(seed, round, position);
+    let byte_index = (position % 256) / 8;
+    let byte = source.get(byte_index).unwrap();
+    let bit = (byte >> (position % 8)) & 1;
+
+    if bit == 1 {
+        flip
+    } else {
+        index
+    }
+}
+
+fn hash_with_round_and_position(seed: B256, round: u8, position: usize) -> Hash256 {
+    let mut context = Context::new();
+
+    context.update(seed.as_slice());
+    context.update(&[round]);
+
+    context.update(&(position / 256).to_le_bytes()[0..4]);
+
+    let digest = context.finalize();
+    Hash256::from_slice(digest.as_ref())
+}
+
+/// Return from ``indices`` a random index sampled by effective balance
+pub fn compute_proposer_index(
+    state: BeaconState,
+    indices: &[u64],
+    seed: B256,
+) -> anyhow::Result<u64> {
+    ensure!(!indices.is_empty(), "Index must be less than index_count");
+
+    let mut i: u64 = 0;
+    let total = indices.len() as u64;
+
+    loop {
+        let shuffled_index = compute_shuffled_index((i % total) as usize, total as usize, seed);
+        ensure!(shuffled_index.is_ok(), "Shuffled index failed");
+
+        let candidate_index = indices[shuffled_index.unwrap()];
+
+        let seed_with_index = [seed.as_slice(), &((i / 32).to_be_bytes())[..]].concat();
+        let hash = hash(&seed_with_index);
+        let random_byte = hash[(i % 32) as usize];
+
+        let effective_balance = state.validators[candidate_index as usize].effective_balance;
+
+        if (effective_balance * MAX_RANDOM_BYTE) >= (MAX_EFFECTIVE_BALANCE * random_byte as u64) {
+            return Ok(candidate_index);
+        }
+
+        i += 1;
+    }
+}
+
+/// Return the committee corresponding to ``indices``, ``seed``, ``index``, and committee ``count``.
+pub fn compute_committee(indices: &[u64], seed: B256, index: u64, count: u64) -> Vec<u64> {
+    let start = (indices.len() as u64 * index) / count;
+    let end = (indices.len() as u64 * (index + 1)) / count;
+
+    (start..end)
+        .map(|i| {
+            let shuffled_index = compute_shuffled_index(i as usize, indices.len(), seed);
+            indices[shuffled_index.unwrap()]
+        })
+        .collect()
+}
```

Comment:
```suggestion
pub fn compute_committee(indices: &[u64], seed: B256, index: u64, count: u64) -> anyhow::Result<Vec<u64>> {
    let start = (indices.len() as u64 * index) / count;
    let end = (indices.len() as u64 * (index + 1)) / count;

    Ok((start..end)
        .map(|i| {
            indices[compute_shuffled_index(i as usize, indices.len(), seed)?]
        })
        .collect())
}
```

don't use unwrap in production code

Reply:
Missed out on removing this `unwrap`. Will make sure it doesnt happen again